### ISSUE TITLE
Depth and Pose on Image with Black Borders

### ISSUE THREAD

**Issue by gilmartinspinheiro** [ClementPinard/SfmLearner-Pytorch#24]
Hi! I already posted about this on the original tensorflow implementation, but i woul also like to know your opinion on the matter :)

Do you think that the black borders on an image like the one bellow would affect the training and the predicitons of depth/pose?

**Comment by ClementPinard** [comment]
I think it will pose some problems because with warping you will compare image parts with black areas. The optimizer will try to avoid it to minimize photometric error.

To my mind, your best solution is to weight the photometric error for each pixel. If you know exactly the pixels to dismiss, just multiply these photometric loss values by 0

You can find an example of how I did the same thing with out of bound warped pixels in `loss_functions.py`

**Comment by gilmartinspinheiro** [comment]
Thank you for your answer! :)
That makes a lot of sense. I will implement it and get back to you as soon as possible!

**Comment by lliuz** [comment]
Hi, @ClementPinard 
I have carefully read how you did for the out of bound warped pixels in `loss_functions.py`, but I am wondering when the pose or depth is not very accurate, such as at the begin of training, there might be a lot of pixels out of bound. Since we multiply these photometric loss values by 0, the photometric loss will be quite small. I think it will lead the network to divergence. 
I have this problem when I use it in other scenarios. Do you have any experience dealing with it?

**Comment by ClementPinard** [comment]
What scenario are you using it on ? A known problem is when translation is not enough, then the depth has very little influence on the warping since the parallax is low.

You can try the `train_flexible_shift.py` script which tries to overcome this problem by increasing temporal shift between frames when the translation predicted is too low.

**Comment by lliuz** [comment]
Thank you for your prompt reply. I think you are right and I have fixed my problem. The divergence in my work is just because of a stupid typo.
